### ISSUE TITLE
Problem matcher errors are unhelpful

### ISSUE THREAD

**Issue by justinfagnani** [microsoft/vscode#192811]
Filing to re-open https://github.com/microsoft/vscode/issues/29951 which either isn't fixed, or is fixed only for some output that I'm not seeing. I cannot find the kind of output referenced in https://github.com/microsoft/vscode/issues/29951#issuecomment-968970049

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.82.0
- OS Version: macOS 13.4.1

Steps to Reproduce:

1. Write a task with a custom problemMatcher, like so:
      ```json
       {
	    "version": "2.0.0",
	    "tasks": [
		    {
			    "label": "build-x",
			    "type": "shell",
			    "command": "npm",
			    "args": [
				    "run",
				    "build",
				    "--watch",
				    "-w",
				    "@lit-labs/x"
			    ],
			    "problemMatcher": [{
				    "background": {
					    "activeOnStart": true,
					    "endsPattern": "Watching for file changes"
				    }
			    }, "$tsc-watch"],
			    "isBackground": true,
			    "presentation": {
				    "reveal": "always"
			    },
			    "group": {
				    "kind": "build",
				    "isDefault": true
			    }
		    }
	    ]
    }
      ```
2. See an error toast with the text: "There are task errors. See the output for details."
3. Click on "Show Output"
4. See the following buffer:

```
Error: the description can't be converted into a problem matcher:
{
    "background": {
        "activeOnStart": true,
        "endsPattern": "Watching for file changes"
    }
}
```

This error message doesn't say what's wrong with the problemMatcher (all fields are optional in the [schema](https://code.visualstudio.com/docs/editor/tasks-appendix), so nothing is obvious to me).

**Comment by justinfagnani** [comment]
If it try to add some more fields to `problemMatcher` I eventually get a different, and maybe more detailed, message, but it's also not helpful:

This
```json
			"problemMatcher": [{
				"owner": "wireit",
				"source": "Wireit",
				"severity": "error",
				"pattern": {
					"regexp": ""
				},
				"background": {
					"activeOnStart": true,
					"beginsPattern": "",
					"endsPattern": "Watching for file changes"
				}
			}, "$tsc-watch"],
```

give me the message:
```
A problem matcher must define both a begin pattern and an end pattern for watching.
```

Is this talking about this `background` matcher? That does have `beginsPattern` and `endsPattern`. The main `problemMatcher` has  `regexp`, but no fields for begin or end.

**Comment by meganrogge** [comment]
does the documentation help you? https://code.visualstudio.com/docs/editor/tasks#_defining-a-problem-matcher

**Comment by justinfagnani** [comment]
@meganrogge not really - it doesn't talk about exactly what's required in a problemMatcher. Combined with all fields int he schema being options and the unhelpful error message it made this a bad experience.

**Comment by ZhenshengLee** [comment]
What's the current status in the year 2025?

**Comment by yury-s** [comment]
Yeah, having a bunch of errors like below, from an old `tasks.json`. It's very hard to tell based on the error what VS Code doesn't like about the config:

```json
Error: the description can't be converted into a problem matcher:
{
    "base": "$gcc",
    "fileLocation": [
        "relative",
        "/home/yurys/webkit/WebKitBuild/WPE/Release"
    ]
}
```

**Comment by floodkoff** [comment]
@yury-s i have exactly the same problem matcher snippet causing the error (the path  is "${workspaceFolder}/../../.."), did you find how to fix it?

**Comment by yury-s** [comment]
@floodkoff Don't remember what it was and I don't see the error anymore, sorry.
